saleor/graphql/discount/mutations/utils.py

```diff
@@ -24,3 +25,11 @@ def convert_catalogue_info_to_global_ids(
             for id_ in catalogue_info[catalogue_field]
         )
     return converted_catalogue_info
+
+
+def clear_promotion_old_sale_id(promotion: Promotion, *, save=False):
+    """Clear the promotion `old_sale_id` if set."""
+    if promotion.old_sale_id:
+        promotion.old_sale_id = None
+    if save:
+        promotion.save(update_fields=["old_sale_id"])
```

Comment:
In case when we receive `save=True`:
 It means that we always make a save action here. Whether you use this function a second or tenth time for a single promotion, we will always make a save.

Reply:
Good point, will change